ha/core/controllers/pcs/node_controller.py

```diff
@@ -260,3 +314,70 @@ def shutdown(self, nodeid: str) -> dict:
                 status: Succeeded, Failed, InProgress
         """
         raise HAUnimplemented("This operation is not implemented.")
+
+    @controller_error_handler
+    def stop(self, node_id: str, timeout: int = -1, **op_kwargs) -> dict:
+        """
+        Stop (poweroff) node with node_id.
+        Args:
+            node_id (str): Private fqdn define in conf store.
+        Returns:
+            ([dict]): Return dictionary. {"status": "", "msg":""}
+                status: Succeeded, Failed, InProgress
+        """
+        check_cluster = op_kwargs.get("check_cluster") if op_kwargs.get("check_cluster") is not None else True
+        poweroff = op_kwargs.get("poweroff") if op_kwargs.get("poweroff") is not None else False
+        storageoff = op_kwargs.get("storageoff") if op_kwargs.get("storageoff") is not None else False
+        try:
+            # Raise exception if node_id is not valid
+            self._is_node_in_cluster(node_id=node_id)
+
+            # Get the nodeid from pvtfqdn
+            nodeid = self._health_manager.get_key(f"{const.PVTFQDN_TO_NODEID_KEY}/{node_id}")
```

Comment:
can we use
confstore..get_key(f"{const.PVTFQDN_TO_NODEID_KEY}/{node_id}")

Reply:
We can use confstore, but in confstore we get dict(key, value) pair which need to be parsed to get values.
But in _health_manager we get "value" directly.

Reply:
yes we can use like
_, nodeid = confstore..get_key(f"{const.PVTFQDN_TO_NODEID_KEY}/{node_id}")
remove _health_manager from pr

Reply:
Yes getting values from confstore.